src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/NativeTypeTableFilter.java

```diff
@@ -108,11 +108,10 @@ public <StmtT extends OngoingWhereClause<StmtT>> StmtT apply(
     try {
       var codec =
           JSONCodecRegistry.codecToCQL(
-              tableSchemaObject.tableMetadata, CqlIdentifier.fromCql(path), columnValue);
+              tableSchemaObject.tableMetadata, getPathAsCqlIdentifier(), columnValue);
       positionalValues.add(codec.toCQL(columnValue));
     } catch (UnknownColumnException e) {
-      // TODO AARON - Handle error
-      throw new RuntimeException(e);
+      throw ErrorCode.TABLE_COLUMN_UNKNOWN.toApiException(e.getMessage());
```

Comment:
Changes 500 SERVER_INTERNAL into proper failure when referencing unknown column in filter, projection etc clauses.